Api/user crud and books

juanmbellini opened the pull request:
# Summary

This PR adds the REST layer for User' CRUD and Book relationship operations, according to the trello card: [https://trello.com/c/m1JwLGxK](https://trello.com/c/m1JwLGxK)

# Notes

The endpoints for the User-Book relationship are:

| Operation                          | Endpoint                                |
|:-------------------------|:----------------------------|
| Get user's books              | /users/:userId/book               |
| Add book to user             | /users/:userId/book/:bookId |
| Remove book from user  | /users/:userId/book/:bookId |

The base branch is repositories/user, which should be changed to dev once [the repositories/user PR](https://github.com/wolox-training/jmb-java/pull/7) is merged.

# Trello Card

https://trello.com/c/m1JwLGxK

juanmbellini committed:
Add a basic CRUD REST API for Users

- Inclde an UserCreationRequestDTO with the necessary fields to create a User:
	- Basically, it contains almost all the User fields, except for the id
	- It also contains Jackson mapping annotations

juanmbellini committed:
Add books add/remove capabilities to UserController

juanmbellini committed:
Add exception handling for BookAlreadyOwnedException

- This allows handling the exception in a clean way (without adding code to the controller method)

juanmbellini committed:
Add missing @Transactional annotations

- Add the annotation to the create and the delete operations

juanmbellini committed:
Change the entity returned by the user creation method

- Return the Download DTO (does not include the books field)

darkBasic666 commented on src/main/java/wolox/training/web/dtos/deserializers/IsoLocalDateDeserializer.java:
```diff
@@ -0,0 +1,32 @@
+ * from ISO 8601 format {@link String}s.
+ */
+public class IsoLocalDateDeserializer extends StdDeserializer<LocalDate> {
+
```
Why do you use deserializer and serializer class?  maybe an annotation is more simple to code

juanmbellini replied:
Yes, I just discovered ```@JsonFormat(pattern = "yyyy-MM-dd")```.
I also discovered that it also works without setting any annotation, which default to ```yyyy-MM-dd```.

juanmbellini committed:
Remove serializers/deserializers

- Apparently, Jackson can serialize/deserialize LocalDate instances using the ISO 8601 format
